clarity/evaluator/haspi/ebm.py

```diff
@@ -1,302 +1,357 @@
 
 import numpy as np
 from scipy.signal import convolve, convolve2d
 
 
-def EnvFilt(xdB, ydB, fcut, fsub, fsamp):
+def env_filter(reference_db, processed_db, filter_cutoff, freq_sub_sample, freq_samp):
     """
-    Function to lowpass filter and subsample the envelope in dB SL produced
-    by the model of the auditory periphery. The LP filter uses a von Hann
-    raised cosine window to ensure that there are no negative envelope values
-    produced by the filtering operation.
+    Lowpass filter and subsample the envelope in dB SL produced by the model
+    of the auditory periphery. The LP filter uses a von Hann raised cosine
+    window to ensure that there are no negative envelope values produced by
+    the filtering operation.
 
     Args:
-        xdB (np.ndarray): env in dB SL for the ref signal in each auditory band
-        ydB (np.ndarray): env in dB SL for the degraded signal in each auditory band
-        fcut ():  LP filter cutoff frequency for the filtered envelope, Hz
-        fsub ():  subsampling frequency in Hz for the LP filtered envelopes
-        fsamp ():  sampling rate in Hz for the signals xdB and ydB
+    reference_db (np.ndarray): env in dB SL for the ref signal in each auditory band
+    processed_db (np.ndarray): env in dB SL for the degraded signal in each auditory band
+    filter_cutoff ():  LP filter cutoff frequency for the filtered envelope, Hz
+    freq_sub_samp ():  subsampling frequency in Hz for the LP filtered envelopes
+    freq_samp ():  sampling rate in Hz for the signals xdB and ydB
 
     Returns:
-        tuple: xLP - LP filtered and subsampled reference signal envelope
-           Each frequency band is a separate column. yLP - LP filtered and subsampled
-           degraded signal envelope
+    tuple: reference_env - LP filtered and subsampled reference signal envelope
+           Each frequency band is a separate column.
+           processed_env - LP filtered and subsampled degraded signal envelope
 
 
+    Updates:
     James M. Kates, 12 September 2019.
     Translated from MATLAB to Python by Zuzanna Podwinska, March 2022.
     """
     # Check the filter design parameters
-    assert fsub <= fsamp, "Error in ebm.EnvFilt: Subsampling rate too high."
-    assert fcut <= 0.5 * fsub, "Error in ebm.EnvFilt: LP cutoff frequency too high."
+    assert (
+        freq_sub_sample <= freq_samp
+    ), "Error in ebm.EnvFilt: Subsampling rate too high."
+    assert (
+        filter_cutoff <= 0.5 * freq_sub_sample
+    ), "Error in ebm.EnvFilt: LP cutoff frequency too high."
 
     # Check the data matrix orientation
     # Require each frequency band to be a separate column
-    nrow = xdB.shape[0]  # number of rows
-    ncol = xdB.shape[1]  # number of columnts
+    nrow = reference_db.shape[0]  # number of rows
+    ncol = reference_db.shape[1]  # number of columnts
     if ncol > nrow:
-        xdB = xdB.T
-        ydB = ydB.T
-    nsamp = xdB.shape[0]
+        reference_db = reference_db.T
```

Comment:
Not sure I like this :( -- seems to be quietly fixing a user-error without so much as a warning. But think best to leave it for now. We need a coherent strategy for logging throughout the code base.

Reply:
Captured in #155